state/postgresql/postgresdbaccess.go

```diff
@@ -389,11 +408,20 @@ func (p *PostgresDBAccess) doDelete(parentCtx context.Context, db dbquerier, req
 }
 
 func (p *PostgresDBAccess) BulkDelete(parentCtx context.Context, req []state.DeleteRequest) error {
-	tx, err := p.db.BeginTx(parentCtx, nil)
+	ctx, cancel := context.WithTimeout(parentCtx, 30*time.Second)
+	tx, err := p.db.Begin(ctx)
+	cancel()
 	if err != nil {
 		return fmt.Errorf("failed to begin transaction: %w", err)
 	}
-	defer tx.Rollback()
+	defer func() {
+		rollbackCtx, rollbackCancel := context.WithTimeout(parentCtx, 30*time.Second)
+		rollbackErr := tx.Rollback(rollbackCtx)
+		rollbackCancel()
+		if rollbackErr != nil && !errors.Is(rollbackErr, pgx.ErrTxClosed) {
+			p.logger.Errorf("Failed to rollback transaction in BulkDelete: %v", rollbackErr)
```

Comment:
should we let the user control the context timeouts if any?

Reply:
Sorry, what was the consensus here? Shouldn't we have a metric representing rollback failures, as it can be an important aspect for user to look into.

Reply:
I am not sure what conditions would cause the rollback to fail. aside from the transaction already being closed (which is an error we expect to get here and it's fine), I can only think of a network issue. In that case, however, if the connection is dropped while a transaction is active (and un-committed), then it's automatically rolled back.

Reply:
Worth adding also that this behavior isn't new and the current code behaves the same way - except there errors were completely ignored and now are at least logged.